cf-Invalidation-service/index.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+/*
+*
+*   Title : Invalidation Service for Covid Alert QA Tools
+*   Purpose : Receive an API call to trigger invalidation.
+*   Author : Timothy Patrick Jodoin
+*   Title : Principal Solution Architect
+*   Email : tj@fsdcsolutions.com
+*   Firm : FSDC Solutions Inc.
+*   Date : Jan 12th, 2021
+*   Client : CDS
+*
+*/
+
+const
+    AWS = require( 'aws-sdk' ),
+    uuid = require( 'uuid' ),
+    CF = new AWS.CloudFront();
+    AWS.config.update({region: 'ca-central-1'});
+
+exports.handler = async event => {
+
+    var distribution_ID = process.env.distributionID;   //encrypted env_var
+    let callerReference = uuid.v1();  // RFC4122 standard uuid Timestamp based generation
+
+// populate invalidation object
+    console.log("ID :",distribution_ID);
+    let distribution = {
+        DistributionId: distribution_ID,
+        InvalidationBatch: {
+            CallerReference: callerReference,                 //unique caller reference id to prevent repeat call attack
+            Paths: {
+                Quantity: 1,
+                Items: [
+                    '/',                                //invaldate root of folder
+                ]
+            }
+        }
+    };
+    try {
+        let data = await CF.createInvalidation(distribution);
+        console.log('INVALIDATION_SUCCESS :', data);
+        return 'INVALIDATION_SUCCESS';
+    }
+    catch (err) {
+        console.log(err, err.stack); // an error occurred
```

Comment:
I imagine if this failed it should be logged as an error executing instead of just logging to console.log.

Reply:
With out this consol.log you will receive no error info of the failed event.

Reply:
Okay I see what is happening here, because you are catching the potential error logging it and returning Invalidation Failed. 

I think it would make more sense to let the exception get thrown and caught by the AWS Framework that runs the lambdas and so it will be exposed through the AWS Tooling that is available for serverless and the lambda will return a proper I'm assuming 5xx code, instead of a 2xx. 

The way this is written I'm pretty sure the lambda will return a success even if it fails.

Reply:
Okay after understanding that errors have to be mapped in API Gateway this is okay.